cdm/src/main/java/ucar/nc2/StringKeeper.java

```diff
@@ -0,0 +1,43 @@
+package ucar.nc2;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * A list of strings that only allows one thread to use any given value at the same time.
+ *
+ * @author cmrose
+ */
+
+public class StringKeeper {
```

Comment:
This is really a type of Lock (see https://docs.oracle.com/javase/tutorial/essential/concurrency/locksync.html and https://winterbe.com/posts/2015/04/30/java8-concurrency-tutorial-synchronized-locks-examples/)  - perhaps StringLock?

Reply:
It is now called StringLocker